sourcegraph/sourcegraph
# blob: do not hijack search input focus

Adds extra condition to CodeMirror extension responsible for keeping the selected occurrence focused to avoid hijacking blob search input focus.

https://user-images.githubusercontent.com/25318659/221198420-cdfeb22b-fa20-4fd9-a32f-fad03615223e.mov


## Test plan
- open a blob that doesn't fit one screen (e.g., `ReactFiberCommitWork.js`)
- open blob search (`Cmd+F`) and search for a string that appears in different parts of the blob (e.g., `ref` if you opened the file mentioned above)
- press `Enter`/`Shift+Enter` to navigate through search matches 
- ensure that search input remains focused
- focus the blob content and use arrow keys to navigate through occurrences 
- use `Cmd+Shift+ArrowDown` to select text to the end of the document
- ensure code editor remains focused and you can use arrow keys to navigate through occurrences.

Tested manually (video attached).




## App preview:

- [Web](https://sg-web-taras-yemets-cm-blob-search-blur.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
